src/MinHeap.js

```diff
@@ -0,0 +1,104 @@
+var MinHeap = function() {
+  var values = [null];
+
+  function getParentIndex(childIndex) {
+    return Math.floor(childIndex / 2);
+  }
+
+  function getChildIndices(parentIndex) {
+    var leftChild = parentIndex * 2;
+
+    return {
+      leftChild: leftChild,
+      rightChild: leftChild + 1,
+    };
+  }
+
+  function swap(firstIndex, secondIndex) {
+    var firstValue = values[firstIndex],
```

Comment:
Can't you use only one `temp` to swap values? Besides readability, any other reason to do it this way?

Reply:
> Besides readability, any other reason to do it this way?

Nope - I find `firstValue` / `secondValue` clearer than using a variable named `temp` or something. I'm open to changing though - thoughts on the memory / readability tradeoff?

Reply:
I like to avoid extra assignments because of all the extra CPU cycles and stack manipulation that happens. In non-critical paths I wouldn't bother much, but in a function that would be called so many times, it might make a difference.